File: quiz-fe/src/share/services/user_question_answer/user-question-answer.service.ts
import { ENV } from '@/share/utils/env';
import { BaseService, PagingResponse } from '../BaseService';
import {
  UserQuestionAnswer,
  UserQuestionAnswerRequest,
  UserQuestionAnswerResponse,
  UserQuestionAnswerView,
} from './models';


export class UserQuestionAnswerService extends BaseService<UserQuestionAnswer, UserQuestionAnswerRequest, UserQuestionAnswerResponse, UserQuestionAnswerView> {
  constructor() {
    super(ENV.API_URL, 'user-question-answers');
  }

  async listPaged(page = 0, size = 10): Promise<PagingResponse<UserQuestionAnswerResponse>> {
    return this.findAllPaged({ page, size });
  }

  async createAnswer(request: UserQuestionAnswerRequest): Promise<UserQuestionAnswerResponse> {
    return this.create(request);
  }
}

export const userQuestionAnswerService = new UserQuestionAnswerService();

File: quiz/src/main/java/com/example/quiz/service/user_question_answer/UserQuestionAnswerService.java
package com.example.quiz.service.user_question_answer;

import com.example.quiz.base.baseInterface.BaseService;
import com.example.quiz.model.entity.user_question_answer.UserQuestionAnswer;
import com.example.quiz.model.entity.user_question_answer.UserQuestionAnswerRequest;
import com.example.quiz.model.entity.user_question_answer.UserQuestionAnswerView;
import com.example.quiz.model.entity.user_question_answer.UserQuestionAnswerResponse;

public interface UserQuestionAnswerService extends BaseService<UserQuestionAnswer, Long, UserQuestionAnswerRequest, UserQuestionAnswerResponse, UserQuestionAnswerView> {
}
